content/en/integrations/guide/service-account-impersonation.md

```diff
@@ -0,0 +1,111 @@
+
+* Service account impersonation and automatic project discovery rely on having certain roles and APIs enabled to monitor projects. Before you start, ensure the following APIs are enabled for the projects you want to monitor:
+  * [Cloud Resource Manager API][3]
+  * [Google Cloud Billing API][4]
+  * [Cloud Monitoring API][5]
+  * [Compute Engine API][6]
+  * [Cloud Asset API][7]
+  * [IAM API][8]
+
+### Setup
+
+### 1. Create your Google Cloud service account
+
+1. Open your [Google Cloud console][9].
+2. Navigate to **IAM & Admin** > **Service Accounts**.
+3. Click on **Create service account** at the top.
+4. Give the service account a unique name, then click **Create and continue**.
+5. Add the following roles to the service account:
+   * Monitoring Viewer
+   * Compute Viewer
+   * Cloud Asset Viewer
+   * Browser
+6. Click **Continue**, then **Done** to complete creating the service account.
+
+{{< img src="integrations/guide/service_account_impersonation/create-service-account.png" alt="Google Cloud console interface, showing the 'Create service account' flow. Under 'Grant this service account access to project', the four roles in the instructions are added." style="width:70%;">}}
+
+### 2. Add the Datadog principal to your service account
+
+1. In Datadog, navigate to the [**Integrations** > **Google Cloud Platform**][10].
+2. Click on **Add GCP Account**. If you have no configured projects, you are automatically redirected to this page.
+3. If you have not generated a Datadog principal for your org, click the **Generate Principal** button.
+4. Copy your Datadog principal and keep it for the next section.
+   {{< img src="integrations/guide/service_account_impersonation/datadog-principal.png" alt="Datadog interface, showing the 'Add New GCP Account' flow. The first step, 'Add Datadog Principal to Google,' features a text box where a user can generate a Datadog Principal and copy it to their clipboard. The second step, 'Add Service Account Email,' features a text box that the user can complete in section 3." style="width:70%;">}}
+   Keep this window open for the [next section](#3-complete-the-integration-setup-in-datadog).
+5. In [Google Cloud console][9], under the **Service Acounts** menu, find the service account you created in the [first section](#1-create-your-google-cloud-service-account).
+6. Go to the **Permissions** tab and click on **Grant Access**.
+   {{< img src="integrations/guide/service_account_impersonation/grant-access.png" alt="Google Cloud console interface, showing the Permissions tab under Service Accounts." style="width:70%;">}}
+7. Paste your Datading principal into the **New principals** text box.
+8. Assign the role of **Service Account Token Creator** and click **Save**.
+   {{< img src="integrations/guide/service_account_impersonation/add-principals.png" alt="Google Cloud console interface, showing an 'Add principals' box and an 'Assign roles' interface." style="width:70%;">}}
```

Comment:
Same here. If we can find one which doesn't have specific principal values or "staging", it would be nicer.

Reply:
@dangteralyn for visibility